### DISCUSSION TITLE
request header propagation

### DISCUSSION THREAD

**Discussion by madhon** [dotnet/yarp#1896] [Q&A] [answered]
I want to add in request header propagation as per the following link, and have it pass down to the clusters, how would i go about getting access to the httpclient used to add the propagation ?

https://thecodeblogger.com/2021/05/25/request-header-propagation-in-net-core-web-applications/

**Comment by Tratcher** [comment, marked as answer]
Microsoft.AspNetCore.HeaderPropagation is redundant with YARP's functionality and the two shouldn't be used together. YARP propagates most headers by default, excluding only a few connection specific ones. Is something not being forwarded that you'd expect? Or did you want to manipulate the forwarded headers? You can look at transforms for that. https://microsoft.github.io/reverse-proxy/articles/transforms.html

**Reply by pastrasigns** [reply]
url doesn't work

**Reply by MihaZupan** [reply]
It does now (see https://github.com/dotnet/yarp/issues/2737 for details).
